docker/web-and-data/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# This image includes both the web app and the data import related scripts in
+# one image. Because of historic reasons some of these scripts run java
+# artifacts under the hood. It therefore currently makes more sense to have a
+# single image supporting both instead of moving the scripts out to a separate
+# container.
+FROM maven:3.5.4 as build
+COPY $PWD /cbioportal
+WORKDIR /cbioportal
+RUN mvn -DskipTests clean install
+
+FROM shipilev/openjdk-shenandoah:8
+RUN mkdir -p /cbioportal
+COPY --from=build /cbioportal/portal/target/cbioportal*.war /app.war
+COPY --from=build /cbioportal/portal/target/dependency/webapp-runner.jar /webapp-runner.jar
+# copy over core jar and scripts
+COPY --from=build /cbioportal/core /cbioportal/core
+COPY --from=build /cbioportal/scripts /cbioportal/scripts
+COPY --from=build /cbioportal/requirements.txt /cbioportal/requirements.txt
+
+# install build and runtime dependencies
+# ignore update failures
+RUN apt-get update; apt-get install -y --no-install-recommends \
+        build-essential \
+        libmysqlclient-dev \
+        python3 \
+        python-dev \
+        python3-dev \
+        python3-pip \
+	&& rm -rf /var/lib/apt/lists/* \
+    && pip3 install -r /cbioportal/requirements.txt
+
+# add importer scripts to PATH for easy running in containers
+RUN find /cbioportal/core/src/main/scripts/ -type f -executable \! -name '*.pl'  -print0 | xargs -0 -- ln -st /usr/local/bin
+
+# put config files in this folder if you want to override config
+ENV PORTAL_HOME=/cbioportal
+
+# TODO: fix the workdir-dependent references to '../scripts/env.pl' and do this:
+# RUN find $PWD/core/src/main/scripts/ -type f -executable \! \( -name env.pl -o -name envSimple.pl \)  -print0 | xargs -0 -- ln -st /usr/local/bin
```

Comment:
can we remove line 38,39 if is no use?

Reply:
And why _is_ it commented out? In [thehyve/cbioportal-docker](https://github.com/thehyve/cbioportal-docker/blob/v2.1.1+backport6012/docs/example_commands.md#importing-data), it's actively used to run the study import scripts.

Reply:
what is line 33 for?

Reply:
@fedde-s that's copy pasted from the Dockerfile in the master branch: https://github.com/thehyve/cbioportal-docker/blob/master/Dockerfile#L57-L58

Reply:
also there's a `TODO` comment in there so figured it might be important to keep?

Reply:
@khzhu line 33 adds scripts like `metaImport.py` to the `PATH` so you can run the documentation @fedde-s linked without having to specify a full path

Reply:
@inodb ,does line 38,39 try to do the same thing as line 33 except for renaming env.pl?

Reply:
> @fedde-s that's copy pasted from the Dockerfile in the master branch: https://github.com/thehyve/cbioportal-docker/blob/master/Dockerfile#L57-L58

Ahh, yeah, sorry, I think _I_ actually wrote that a long time ago. It's an existing inflexibility in the cBioPortal scripts: the endpoints for loading side-data (say, updating the gene table) only work if you set the working directory to `/cbioportal/core/src/main/scripts`, so I couldn't improve usability by adding them to `$PATH` with my limited knowledge of Perl.

I should report this as an issue in https://github.com/cBioPortal/cbioportal/ instead of tracking it as a TODO comment here. I believe that that would resolve @khzhu's objection.

Reply:
i see this has been reported now: https://github.com/cBioPortal/cbioportal/issues/6123